tenacity/wait.py

```diff
@@ -69,27 +69,36 @@ def __call__(self, previous_attempt_number, delay_since_first_attempt_ms):
             self.wait_funcs))
 
 
-class wait_combine_stepped(object):
-    """Combine one or more waiting strategies in a stepped fashion.
+class wait_chain(object):
+    """Chain two or more waiting strategies.
+
+    If all strategies are exhausted, the very last strategy is used
+    thereafter.
 
     For example::
 
-        @retry(wait=wait_combine_stepped(5, wait_fixed(1000), wait_fixed(2000), wait_fixed(5000)))
+        @retry(wait=wait_chain([wait_fixed(1000) for i in range(3)],
+                               [wait_fixed(2000) for j in range(5)],
+                               [wait_fixed(5000) for k in range(4)))
         def wait_stepping():
-            print("Wait 1s for 5 attempts, 2s for 5 attempts and 5s thereafter.")
+            print("Wait 1s for 3 attempts, 2s for 5 attempts and 4s thereafter.")
     """
 
-    def __init__(self, step, *strategies):
-        self.step = int(step)
-        if self.step <= 0:
-            raise ValueError("Step '%s' must be larger than 0." % self.step)
-        self.wait_funcs = list(strategies)
+    def __init__(self, *strategies):
+        self.strategies = []
+        self._add_strategies(*strategies)
+
+    def _add_strategies(self, *strategies):
```

Comment:
Yeah this is a bit hackish, I think it's better to make the caller use + on its list so it can pass them as `wait_chain(*([mylist] + [mylist] …))` or just `wait_chain(wait_fixed(1000), wait_fixed(2000))`